cmd/helm/profiling.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright The Helm Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+	"runtime/pprof"
+	"strings"
+)
+
+var (
+	cpuProfileFile *os.File
+	cpuProfilePath string
+	memProfilePath string
+)
+
+func init() {
+	cpuProfilePath = os.Getenv("HELM_PPROF_CPU_PROFILE")
+	memProfilePath = os.Getenv("HELM_PPROF_MEM_PROFILE")
+}
+
+// startProfiling starts profiling CPU usage if HELM_PPROF_CPU_PROFILE is set
+// to a file path. It returns an error if the file could not be created or
+// CPU profiling could not be started.
+func startProfiling() error {
+	if cpuProfilePath != "" {
+		var err error
+		cpuProfileFile, err = os.Create(cpuProfilePath)
+		if err != nil {
+			return fmt.Errorf("could not create CPU profile: %w", err)
+		}
+		if err := pprof.StartCPUProfile(cpuProfileFile); err != nil {
+			cpuProfileFile.Close()
+			cpuProfileFile = nil
+			return fmt.Errorf("could not start CPU profile: %w", err)
+		}
+	}
+	return nil
```

Comment:
`if memProfilePath != "" {` seems to be missing here (bad merge?)

Reply:
I thought a snapshot was just taken at the end.

Reply:
> I thought a snapshot was just taken at the end.

Correct. Unlike CPU, there is no step where collecting memory profiling data is started. The snapshot can be taken at any time.

Reply:
oops, my bad